packages/react-components/react-alert/src/components/Alert/Alert.types.ts

```diff
@@ -1,15 +1,38 @@
+import { Button } from '@fluentui/react-button';
+
 import type { ComponentProps, ComponentState, Slot } from '@fluentui/react-utilities';
 
 export type AlertSlots = {
-  root: Slot<'div'>;
+  /**
+   * The root slot is the container for the alert component
+   */
+  root: NonNullable<Slot<'div'>>;
+  /**
+   * The icon slot renders the icon determined by the `icon` or `intent` prop
+   */
+  icon?: Slot<'span'>;
+  /**
+   * The content slot renders the text message in the alert component
+   */
+  content: NonNullable<Slot<'span'>>;
+  /**
+   * The action slot renders a button that prompts the user to take action on the alert
+   */
+  action?: Slot<typeof Button>;
 };
 
 /**
  * Alert Props
  */
-export type AlertProps = ComponentProps<AlertSlots>;
+export type AlertProps = ComponentProps<AlertSlots> & {
+  /**
+   * The intent prop, if present, determines the icon to be rendered in the icon slot. It
+   * overrides the value of the icon prop
+   */
+  intent?: 'info' | 'success' | 'error' | 'warning';
+};
 
 /**
  * State used in rendering Alert
  */
-export type AlertState = ComponentState<AlertSlots>;
+export type AlertState = ComponentState<AlertSlots> & Pick<AlertProps, 'intent'>;
```

Comment:
_nit:_ I think you probably want to have intent be required in state.

```suggestion
export type AlertState = ComponentState<AlertSlots> & Required<Pick<AlertProps, 'intent'>>;
```

Reply:
I removed the required piece to make it nullable. May be there's a better way to do this.